torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -109,6 +115,30 @@ std::unique_ptr<RpcCommandBase> deserializeResponse(
     case MessageType::CLEANUP_AUTOGRAD_CONTEXT_RESP: {
       return autograd::CleanupAutogradContextResp::fromMessage(response);
     }
+    case MessageType::RUN_WITH_PROFILING_RESP: {
+      std::unique_ptr<RpcCommandBase> rpcPtr =
+          autograd::RpcWithProfilingResp::fromMessage(response);
+      RpcCommandBase& rpc = *rpcPtr;
+      auto& rpcWithProfilingResp =
+          static_cast<autograd::RpcWithProfilingResp&>(rpc);
+      // read and reconstruct events
+      // Check if the profiler is enabled
+      auto enabled = torch::autograd::profiler::profilerEnabled();
+      TORCH_CHECK(
+          enabled,
+          "Profiler was expected to be enabled. This can happen in callback "
+          " continutations that run in different threads, and the TLS of the "
+          " profiler was not propagated.");
+      std::vector<torch::autograd::profiler::Event> events =
+          rpcWithProfilingResp.getProfiledEvents();
```

Comment:
Do we need a moveProfiledEvents here to avoid copy? It would invalidate `rpcWithProfilingResp` and causing problem for the `moveWrappedRpc` below. Maybe also need sth like `moveAll`?

Reply:
Yes, we could possibly implement a `moveAll` that returns a set of the moved items from the message. In this case, I guess just a pair could do, the events and the wrapped RPC